data-pipeline/src/trace_exporter.rs

```diff
@@ -175,20 +193,40 @@ impl TraceExporter {
                     .await
                 {
                     Ok(response) => {
-                        if response.status() != 200 {
+                        let response_status = response.status();
+                        if response_status != http::StatusCode::OK {
                             let body_bytes = hyper::body::to_bytes(response.into_body()).await?;
                             let response_body =
                                 String::from_utf8(body_bytes.to_vec()).unwrap_or_default();
+                            self.emit_metric(
+                                HealthMetric::Count(STAT_SEND_TRACES_ERRORS, 1),
+                                Some(vec![
+                                    &Tag::new("response_code", response_status.as_str()).unwrap()
```

Comment:
Are we confident that we'll never panic on this unwrap?

Reply:
Docs say `response_status`'s `as_str` is a numeric string:

> The return value only includes a numerical representation of the status code. The canonical reason is not included.

And so yes, it won't panic, since `response_code:{response_status}` is going to be a valid tag name.

Reply:
I'm like 99% confident since this can only really fail if we generate an invalid tag (so like an empty tag, or end with a `:`). The Status code itself is just a `NonZeroU16` so I think functionally speaking, it should be impossible for this to panic.

But since I'm not 100% sure here I'll just add a quick little check and an error log if it ever happens :P

Reply:
@morrisonlevi That makes sense, thanks!